packages/viewer/src/lib/pins/interactions.ts

```diff
@@ -0,0 +1,201 @@
+    element.addEventListener('pointerdown', this.handlePointerDown);
+  }
+
+  public dispose(): void {
+    this.rectObserver.disconnect();
+
+    this.element?.removeEventListener(
+      'pointermove',
+      this.handlePointerMoveCursorCheck
+    );
+    this.element?.removeEventListener('pointerdown', this.handlePointerDown);
+
+    this.element = undefined;
+    this.api = undefined;
+  }
+
+  public async getWorldPositionForPoint(
+    pt: Point.Point
+  ): Promise<Vector3.Vector3 | undefined> {
+    return this.ifInitialized(async ({ api }) => {
+      const vector3 = await api.getWorldPointFromViewport(pt);
+
+      return vector3;
+    });
+  }
+
+  public async handlePlacePin(
+    pt: Point.Point,
+    existingPin?: Pin
+  ): Promise<void> {
+    this.ifInitialized(async ({ api }) => {
+      const [hit] = await api.hitItems(pt);
+
+      if (hit?.hitPoint != null && this.elementRect != null) {
+        const vector3 = await this.getWorldPositionForPoint(pt);
+
+        const relativePoint = translatePointToRelative(
+          {
+            ...pt,
+          },
+          this.elementRect
+        );
+        if (vector3 != null) {
+          const pinId = existingPin != null ? existingPin.id : UUID.create();
+
+          switch (this.controller.getToolType()) {
+            case 'pin':
+              this.controller.setEntity(new DefaultPin(pinId, vector3, pt));
+              break;
+            case 'pin-label':
+              this.controller.setEntity(
+                new TextPin(
+                  pinId,
+                  vector3,
+                  pt,
+                  isTextPin(existingPin)
+                    ? {
+                        ...existingPin.attributes,
+                      }
+                    : {
+                        labelPoint: relativePoint,
+                      },
+                  hit?.itemId?.hex ?? undefined
+                )
+              );
+              break;
+          }
+        }
+      } else {
+        this.controller.setSelectedPinId();
+      }
+    });
+  }
+
+  private handlePointerMoveCursorCheck = async (
+    event: PointerEvent
+  ): Promise<void> => {
+    if (await this.isDroppableSurface(event)) {
+      this.addCursor(this.getCusorType());
+    } else {
+      this.clearCursor();
+    }
+  };
+
+  private getCusorType(): Cursor {
+    switch (this.controller.getToolType()) {
+      case 'pin':
+        return pinCursor;
+      case 'pin-label':
+        return labelPinCursor;
+    }
+  }
+
+  private handlePointerMove = async (event: PointerEvent): Promise<void> => {
+    if (await this.isDroppableSurface(event)) {
+      this.addCursor(this.getCusorType());
+    } else {
+      this.clearCursor();
+    }
+  };
+
+  private handlePointerDown = async (
+    pointerDown: PointerEvent
+  ): Promise<Disposable> => {
+    const pointerUp = (pointerUp: PointerEvent): void => {
+      const distnaceBetweenStartAndEndPoint = Point.distance(
```

Comment:
`distanceBetweenStartAndEndPoint` instead of `distnaceBetweenStartAndEndPoint` 🙂